cypress/integration/networking.spec.ts

```diff
@@ -112,6 +124,127 @@ context("p2p networking", () => {
               "Following"
             )
             .should("exist");
+
+          cy.log("test commit replication from maintainer to contributor");
+
+          cy.log("add a new commit to the project from maintainer's node");
+          const projctPath = path.join(maintainerNodeWorkingDir, projectName);
+          const credentialHelper = `'!f() { test "$1" = get && echo "password=${node1User.passphrase}"; }; f'`;
+          const maintainerCommitSubject =
+            "Commit replication from maintainer to contributor";
+
+          cy.exec(
+            `function commands() {cd ${projctPath} && ` +
+              `touch README.md && ` +
+              `git add . && ` +
+              `git commit -m "${maintainerCommitSubject}" && ` +
+              `git -c credential.helper=${credentialHelper} push rad; }; ` +
+              `PATH=$PWD/proxy/target/release:$PATH commands`,
+            {
+              env: {
+                RAD_HOME: maintainerNode.storagePath,
+              },
+            }
+          );
+
+          cy.log("refresh the UI, because new commits don't show up otherwise");
+          cy.get("body").type("{esc}");
+          commands.pick("sidebar", "profile").click();
+
+          cy.log("make sure new commit shows up in the UI of the maintainer");
+          commands.pick("project-list-entry-new-fancy-project.xyz").click();
+          commands.pick("commits-tab").click();
+          commands
+            .pickWithContent(["commits-page"], maintainerCommitSubject)
+            .should("exist");
+
+          cy.log("check that the commit shows up in the contributor's node");
+          nodeManager.asNode(contributorNode);
+          commands.pick("following-tab").click();
+          commands.pick("project-list-entry-new-fancy-project.xyz").click();
+          commands.pick("commits-tab").click();
+          commands
+            .pickWithContent(["commits-page"], maintainerCommitSubject)
+            .should("exist");
+
+          cy.log("test commit replication from contributor to maintainer");
+
+          const contributorNodeWorkingDir = path.join(
+            tempDirPath,
+            "contributorNode"
+          );
+
+          cy.exec(`mkdir -p ${contributorNodeWorkingDir}`);
+
+          ipcStub.getStubs().then(stubs => {
+            stubs.IPC_DIALOG_SHOWOPENDIALOG.returns(contributorNodeWorkingDir);
+          });
+          commands.pick("checkout-modal-toggle").click();
+          commands.pick("choose-path-button").click();
+          // Make sure UI has time to update path value from stub,
+          // this prevents this spec from failing on CI.
+          cy.wait(500);
```

Comment:
As discussed: We don’t need to wait here.

Reply:
6d50aaabb45f1a3ab92a4737c3c830e8e5d11251